### PR TITLE
Create export account 696

### LINKED ISSUE #696: Create export account data endpoint
**Describe the solution you'd like**
Create a new controller and endpoint to export a user's book data:
- All of the books in their predefined and custom shelves
- All tags, publishers, etc.

**Additional context**
- Be sure to read the contributing guidelines first!
- Discuss your implementation approach with us before we assign you to it
- Branch from `main` and set `main` as the destination branch in your pull request

If you need help with anything, we'll be happy to help you over a [GitHub Q&A discussion](https://github.com/Project-Books/book-project/discussions/categories/q-a) or our [Slack workspace](https://camo.githubusercontent.com/eef8443bce0a2fdc0706ffcd99f91181ac849a4ef423a1ed61386ca0a8b448bb/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f736c61636b2d7465616d626f6f6b70726f6a6563742d3441313534423f6c6f676f3d736c61636b)

This needs to be thoroughly tested.

### PR DESCRIPTION
**Pull request by lrvignesh**
## Summary of change

- Add Book Export API which allows users to export the books data at a point in time and can be used to import here or elsewhere if needed.
- API Endpoint - _**/api/export-books-data**_
- There was a minor issue where the BookGenre Set hasn't been initialized which would throw NPE when we try to add a genre. Fixed that.
- Add a Query method in BookRepository which allows to fetch all the necessary associated details (such as tags, publishers, shelves, etc..) for all the books that a user has added
- Add BookExportController
- Add BookExportService
- Add necessary tests such as
  - BookRepositoryTest (for newly added query)
  - BookExportControllerTest
  - BookExportServiceTest
  - Integration test - BookExportServiceIntegrationTest

## Additional context

Sample JSON Response for exported Book details for a user

```
{
  "books": [
    {
      "title": "Cloud Atlas",
      "numberOfPages": 650,
      "pagesRead": null,
      "bookGenre": [
        "Fantasy"
      ],
      "bookFormat": null,
      "seriesPosition": null,
      "edition": null,
      "bookRecommendedBy": "Loki",
      "isbn": null,
      "yearOfPublication": null,
      "author": {
        "fullName": "David Mitchell"
      },
      "predefinedShelf": {
        "shelfName": "To read"
      },
      "userCreatedShelf": {
        "shelfName": "Favourites"
      },
      "tags": [
        {
          "name": "Epic"
        }
      ],
      "publishers": [
        {
          "name": "Sceptre"
        }
      ],
      "rating": null,
      "dateStartedReading": null,
      "dateFinishedReading": null,
      "bookReview": null
    }
  ]
}
```

## Related issue

Closes #696

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request

- [x] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

For any of the optional checkboxes (e.g. the screenshots one), still check it if it does not apply.

### CODE REVIEW

File: backend/book-app/src/main/java/com/karankumar/bookproject/model/Book.java [hunk 1]
```diff
@@ -89,7 +89,7 @@
   @ElementCollection(targetClass = BookGenre.class, fetch = FetchType.EAGER)
   @CollectionTable(name = "book_genre", joinColumns = @JoinColumn(name = "book_id"))
   @Column(name = "genre")
-  private Set<BookGenre> bookGenre;
+  private Set<BookGenre> bookGenre = new HashSet<>();
```
Comment (knjk04) [inline backend/book-app/src/main/java/com/karankumar/bookproject/model/Book.java:92]: Good spot, thanks for adding this!

File: backend/book-app/src/test/java/com/karankumar/bookproject/bookexport/BookExportServiceIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
+
+@IntegrationTest
+@Slf4j
+class BookExportServiceIntegrationTest {
```
Comment (knjk04) [inline backend/book-app/src/test/java/com/karankumar/bookproject/bookexport/BookExportServiceIntegrationTest.java:47]: Could you please remove this? We're moving towards only having integration tests at the controller level

**Review by knjk04 — CHANGES_REQUESTED** [review]
Thanks a lot for adding the export feature! This works well and was very well written!

I've left one small comment about not needing the integration test at the service level

File: backend/book-app/src/test/java/com/karankumar/bookproject/bookexport/BookExportServiceIntegrationTest.java (hunk 2 above)
Comment (lrvignesh) [reply backend/book-app/src/test/java/com/karankumar/bookproject/bookexport/BookExportServiceIntegrationTest.java]: Hi, I have removed this now

**Review by knjk04 — APPROVED** [review]
Thanks a lot!

**Pull request merged**